### PR TITLE
1038 integrate watcher

### LINKED ISSUE #1038: [watcher, client, node] Add to client/node
Add watcher to client/node and npm publishing scripts.

Should be default `enable`d for the node, and `disable`d for the client

### PR DESCRIPTION
**Pull request by LayneHaber**
## The Problem
Fixes #1038 

## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

**Review by bohendo — APPROVED** [review]
Looks good to me

File: modules/node/migrations/1597733020691-add-watcher.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class addWatcher1597733020691 implements MigrationInterface {
+    name = 'addWatcher1597733020691'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`DELETE FROM "typeorm_metadata" WHERE "type" = 'VIEW' AND "schema" = $1 AND "name" = $2`, ["public","anonymized_onchain_transaction"]);
+        await queryRunner.query(`DROP VIEW "anonymized_onchain_transaction"`);
```
Comment (rhlsthrm) [inline modules/node/migrations/1597733020691-add-watcher.ts:8]: Is this doing anything?

File: modules/node/src/challenge/challenge.provider.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  async initiateChallenge(
+    pubId: string,
+    chainId: number,
+    data: { multisigAddress: string; appIdentityHash: string },
+  ) {
+    return this.challengeService.initiateChallenge(data.appIdentityHash, data.multisigAddress);
+  }
+
+  async setupSubscriptions(): Promise<void> {
```
Comment (rhlsthrm) [inline modules/node/src/challenge/challenge.provider.ts:48]: I'm not understanding why this interface needs to be exposed to the client at all? Why is the client telling the node to cancel/initiate challenges?

Comment (rhlsthrm) [reply modules/node/src/challenge/challenge.provider.ts:48]: If it's a debugging thing, it should be exposed through the REST interface.

**Review by rhlsthrm — APPROVED** [review]

File: modules/node/migrations/1597733020691-add-watcher.ts (hunk 1 above)
Comment (bohendo) [reply modules/node/migrations/1597733020691-add-watcher.ts:8]: At some point a view was removed from one of the entities & this is bringing the db schema up to date w that change

**Pull request merged**